src/lib/blueprint.ts

```diff
@@ -1,7 +1,33 @@
 import type { Openapi } from './openapi.js'
 
+interface Parameter {
+  name: string
+  type: string
+  required: boolean
+  description: string
+}
+
+interface ReturnValue {
+  type: string
+  description: string
+}
+
+interface Example {
+  codeSnippet: string
+  explanation: string
+}
+
 export interface Blueprint {
   name: string
+  routes: {
+    path: string
+    method: string
+    methodDescription: string
+    category: string
```

Comment:
Thought it might be good to make the blueprint resemble our API reference and possibly list all methods under specific categories like we do in docs, e.g. category: Connect Webviews that groups all methods done by calling connect_webviews/*
Might be too much of a hassle though for not a lot of gain

Reply:
No, this is really important! But the category as a string is not enough. We should define them properly. 

This is similar to the `namespace` property [defined here](https://github.com/seamapi/javascript-http/blob/bea6d987a58597692dc6b3640c39f953a979f819/generate-routes.ts#L92) and related to the tags in the openapi spec.

Note that the blueprint does NOT have to be efficient for storage, it should be efficient for usage, so IMO duplicating data is ok, e.g., we can do

```suggestion
interface Route {
  ...
  category: Category
}

interface Category {
  name: string // Devices
  description: string // Manage devices with Seam
  path: string // /devices
}

interface Blueprint {
  routes: Route[]
  categories: Category[]
}
```

Alternatively, we can organize all the routes under categories,

```suggestion
interface Route {
    path: string // /devices/list
}

interface Category {
  name: string // Devices
  description: string // Manage devices with Seam
  path: string // /devices
  routes: Route[]
}

interface Blueprint {
  categories: Category[]
}
```

Reply:
When doing codegen or docgen, do we always do effectively two loops? First over the category, second over the routes in the category?

Reply:
@kainpets Moving this comment into this issue so it's easier to find in the future: https://github.com/seamapi/blueprint/issues/1

Reply:
@kainpets Continuing this comment into this issue so it's easier to find in the future: https://github.com/seamapi/blueprint/issues/1